projects/models.py

```diff
@@ -54,13 +54,12 @@ def is_overdue(self):
 
     def set_to_done(self):
         "Sets all the features associated with it to done, returns false if no features"
-        Project.objects.filter(id=self.id).update(status=30)
+        self.status=Project.ProjectStatus.COMPLETED
+        self.save()
         has_feature = False
-        data_f = Feature.objects.all()
-        Feature.objects.filter(project=self.id).update(status=20)
-        for feat in data_f:
-            if (feat.project.id == self.id):
-                has_feature = True
+        Feature.objects.filter(project=self.id).update(status=Feature.FeatureStatus.DONE)
+        if (Project.objects.get(id=self.id).feature_set.all().count() > 0):
```

Comment:
Again, this is querying the entire database for both Features and Projects, when that information is already available on the `self` object.

If you recall from CS340:
```python
Feature.objects.filter(project=self.id)
```
is equivalent to:
```sql
SELECT * FROM features WHERE id=self.id
```
which is querying the entire database. Same with:

```python
Project.objects.get(id=self.id)
```
is the same as:
```sql
SELECT * FROM projects WHERE id=self.id
```
again querying the entire database.

**The project object is already available as the `self` variable.** This is Object Oriented Programming like we did in CS161/162 except in C++ the variable is called `this`. In Python it is called `self`. So if you're already inside the Project model, the `self` refers to the instance of the Project.

Reply:
This is how to modify the Features that are already associated with the Project (ManyToOne relationship, remember from CS340 with the Foreign Key), without querying the database again.
```suggestion
        for feature in self.feature_set.all():
            feature.status = Feature.FeatureStatus.DONE
            feature.save()
```

Reply:
fixed. Test worked with this change.